doc/modules/linear_model.rst

```diff
@@ -1193,6 +1193,74 @@ Also, this estimator is different from the R implementation of Robust Regression
 squares implementation with weights given to each sample on the basis of how much the residual is
 greater than a certain threshold.
 
+.. _quantile_regression:
+
+Quantile Regression
+===================
+
+Quantile regression estimates median or other quantiles of :math:`y` conditional on :math:`X`, while OLS estimates
+conditional mean.
+
+The :class:`QuantileRegressor` applies linear loss to all samples. It is thus more radical than
+:class:`HuberRegressor`, that applies linear penalty to small fraction of outliers and quadratic loss
+to the rest of observations. :class:`QuantileRegressor` also supports L1 and L2 regularization,
+like :class:`ElasticNet`. It solves
+
+.. math::
+    \underset{w}{min\,} { \frac{1}{n_{samples}} L_q (y - X w) + \alpha \rho ||w||_1 + \alpha(1-\rho) ||w||_2 ^ 2}
+
+where
+
+.. math::
+    \L_q(t) =
+    \begin{cases}
+        q t, & t > 0, \\
+        0,    & t = 0, \\
+        (1-q) t, & t < 0
+    \end{cases}
+
+and :math:`q \in (0, 1)` is the quantile to be estimated.
+
+Quantile regression may be useful if one is interested in predicting an interval
+instead of point prediction. Sometimes prediction interval is calculated based on
+assumption that prediction error is distributed normally with zero mean and constant variance.
+Quantile regression provides sensible prediction intervals even for errors with non-constant
+(but predictable) variance or non-normal distribution.
+
+.. figure:: /auto_examples/linear_model/images/sphx_glr_plot_quantile_regression_001.png
+   :target: ../auto_examples/linear_model/plot_quantile_regression.html
+   :align: center
+   :scale: 50%
+
+Another possible advantage of quantile regression over OLS is its robustness
+to outliers, because it is only sign of an error that influences estimated
+coefficients, not its absolute value.
```

Comment:
shouldn't this be "it is the value [...] rather than the squared error" ? "only the sign of an error" would correspond to L0 loss

Reply:
The model parameters are updated by gradient descent, and the gradient of L1 loss is the sign of the error, while the gradient of L2 loss is twice the error.

The explanation in term of gradients seems more intuitive to me, because it explicitly shows that with L1 loss it does not matter how large the error of a particular data point is - the resulting parameter estimates will be the same, as long as the sign of the error does not change.